src/scripts/main.js

```diff
@@ -1 +1,84 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+let leftClick = false;
+let rightClick = false;
```

Comment:
The variables `leftClick` and `rightClick` are used to track mouse button clicks for the third promise. However, they are never reset, which means that once both buttons have been clicked, the third promise will resolve on any subsequent click. Consider resetting these variables after the promise resolves to avoid this behavior.